backend/src/main/java/com/redhat/cloud/notifications/db/builder/Parameters.java

```diff
@@ -0,0 +1,40 @@
+package com.redhat.cloud.notifications.db.builder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class Parameters {
+
+    private final Map<String, Object> parameters = new HashMap<>();
+
+    // Only used for testing
+    Map<String, Object> mapCopy() {
+        return new HashMap<>(parameters);
+    }
+
+    void forEach(java.util.function.BiConsumer<String, Object> action) {
+        this.parameters.forEach(action);
+    }
+
+    void merge(Parameters other) {
+        other.parameters.forEach(this::addParam);
+    }
+
+    void addParams(Object... params) {
+        if (params.length % 2 != 0) {
+            throw new IllegalArgumentException("Odd number of params. It needs to be a pair number of params. The key followed by the value");
+        }
+
+        for (int i = 0; i < params.length; i += 2) {
+            addParam(params[i].toString(), params[i + 1]);
```

Comment:
If the implementer used it, yes :D, but that would be an invalid usage.
Maybe instead of Objects i could accept a tuple-like object with a `name` (string) and a `value`  (Object)

Reply:
`.ifAnd(someCondition, "foo = :bar", "bar", null)` is an invalid usage indeed, but right now it will throw a cryptic NPE from the line highlighted above while we could check at the entry points for `null` values and forbid them with a nicer exception. This is really not important for the first iteration though, a simple TODO here would be enough.

Reply:
Would it work if added here instead? the stacktrace would point to the entrypoint as this function is called immediatly on all the ifAnd, if, or, ifOr

(updated the code with the exception)